index.ts

```diff
@@ -516,9 +519,40 @@ const getRepoURL = (url?: URL | HTMLAnchorElement | Location): string => {
 	return url.pathname.slice(1).split('/', 2).join('/');
 };
 
+export interface RepositoryInfo {
+	owner: string;
+	name: string;
+	url: string;
+	path: string;
+}
+
+const getRepositoryInfo = (url?: URL | HTMLAnchorElement | Location | string): Partial<RepositoryInfo> => {
+	if (!url) {
+		const canonical = document.querySelector<HTMLMetaElement>('[property="og:url"]'); // `rel=canonical` doesn't appear on every page
+		url = canonical ? canonical.content : location;
+	}
+
+	if (typeof url === 'string') {
+		url = new URL(url, location.origin);
+	}
+
+	if (!isRepo(url)) {
+		return {};
```

Comment:
I decided to return `{}` instead of `undefined` to make it easier to check without having to use `?.`

```js
// Before:
if (getRepoPath() === undefined) 

// Now:
if (getRepositoryInfo().path === undefined)

// If it returned `undefined`:
if (getRepositoryInfo()?.path === undefined)
```

Reply:
Changed my mind. Returning `Partial<RepositoryInfo>` is actually wrong because the type is actually `RepositoryInfo | {}`.

The difference is that with the first one, even if you typecheck `if !repository.name, return`, `.path` will still be `string | undefined`, even if we know for sure it's a string.

Unfortunately `RepositoryInfo | {}` is awkward to use because `repository.name` will cause a type error: `name doesn't exist on {}`. 

In short, since we have `?.` in the language, let's make use of it for ease of use, even if it generates slightly longer code.